internal/controller/api/v1alpha1/image_types.go

```diff
@@ -21,15 +21,19 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	ImageStateCreated        = "Image has been created"
+	ImageStateFailedToCreate = "Failed to create image"
+)
+
 // ImageSpec defines the desired state of Image
 type ImageSpec struct {
 	// The name of the Cluster CR
 	ClusterRef   *corev1.ObjectReference `json:"clusterRef"`
 	SSHPublicKey string                  `json:"sshPublicKey,omitempty"`
 	// The name of the secret containing the pull secret
-	PullSecretRef         *corev1.SecretReference `json:"pullSecretRef"`
-	IgnitionOverrides     string                  `json:"ignitionOverrides,omitempty"`
-	StaticIpConfiguration string                  `json:"staticIpConfiguration,omitempty"`
+	IgnitionOverrides     string `json:"ignitionOverrides,omitempty"`
```

Comment:
https://coreos.slack.com/archives/C01HA8GULPR/p1609919051036300
Because of how rest-api is implemented, it's required to create a cluster, and because we have another cluster api that is using pull secret https://github.com/openshift/assisted-service/blob/master/swagger.yaml#L755
Currently if we have it in both cluster and image with the current backend implementation you will have two resources that will update the same value that is being stored in the cluster.